src/main/java/emissary/util/DateTimeFormatParser.java

```diff
@@ -0,0 +1,206 @@
+                }
+                final DateTimeFormatter sdf = initialSdf;
+                dateFormats.add(sdf);
+            } catch (Exception ex) {
+                logger.debug("{} entry '{}' cannot be parsed", entryType, dfentry, ex);
+            }
+        }
+        logger.debug("Loaded {} {} entries", dateTimeZoneFormats.size(), entryType);
+    }
+
+    /**
+     * Parse an RFC-822 Date or one of the thousands of variants make a quick attempt to normalize the timezone information
+     * and get the timestamp in GMT. Should change to pass in a default from the U124 header
+     *
+     * @param dateString the string date from the RFC 822 Date header
+     * @param supplyDefaultOnBad when true use current date if sentDate is unparseable
+     * @return the GMT time of the event or NOW if unparseable, or null if supplyDefaultOnBad is false
+     */
+    public static LocalDateTime parseDate(final String dateString, final boolean supplyDefaultOnBad) {
+
+        if (dateString != null && dateString.length() > 0) {
+            // Take it apart and stick it back together to
+            // get rid of multiple contiguous spaces
+            String instr = dateString.replaceAll("\t+", " "); // tabs
+            instr = instr.replaceAll("[ ]+", " "); // multiple spaces
+            instr = instr.replaceAll("=0D$", ""); // common qp'ified ending
+
+            LocalDateTime date;
+            date = tryParseWithDateTimeZoneFormats(instr);
+            if (date != null)
+                return date;
+
+            date = tryParseWithDateTimeOffsetFormats(instr);
+            if (date != null)
+                return date;
+
+            date = tryParseWithDateTimeFormats(instr);
+            if (date != null)
+                return date;
+
+            date = tryParseWithDateFormats(instr);
+            if (date != null)
+                return date;
+
+            try {
+                return Instant.from(DateTimeFormatter.ISO_INSTANT.parse(instr)).atZone(GMT).toLocalDateTime();
+            } catch (DateTimeParseException e) {
+                // ignore
+            }
+
+            // If none of these methods worked, use the default if required
+            if (supplyDefaultOnBad) {
+                return LocalDateTime.now();
+            }
+        }
+        return null;
+    }
+
+
+    /**
+     * Attempt to parse the string instr with one of the ZonedDateTime patterns
+     *
+     * @param instr the string to attempt to format
+     * @return the LocalDateTime object if a formatter worked, or null otherwise
+     */
+    private static LocalDateTime tryParseWithDateTimeZoneFormats(final String instr) {
+        // formats with a time zone
+        for (final DateTimeFormatter dtf : dateTimeZoneFormats) {
+            try {
+                ZonedDateTime zdt = ZonedDateTime.parse(instr, dtf);
+                return ZonedDateTime.ofInstant(zdt.toInstant(), GMT).toLocalDateTime();
```

Comment:
It seems like this normalizes to a local time zone? Would we want to add a default time zone for the cases where there is not one? We could assume local or we could assume GMT or really any other depending on the configuration?

Reply:
I can add a configurable option to pick a time zone to assume or use local time.

Reply:
I checked the java documentation and tested myself to confirm: toLocalDateTime should just convert it into a LocalDateTime object and not convert to a local time zone. 
https://docs.oracle.com/javase/8/docs/api/java/time/ZonedDateTime.html#toLocalDateTime--
But I will add the option to choose a time zone to be configurable.